lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -306,11 +306,20 @@ def test_clear_tools_test85(pbench_run, tools_configuration, groups, remotes, to
         remote_check = True
         tool_check = True
         if groups:
-            group_check = any(x in tool_str for x in groups.split(",") if x)
+            groups_list = groups.split(",")
+            group_check = not bool(
+                [group for group in groups_list if not group]
```

Comment:
or `all(groups_list)`??

Reply:
And it would be good to have a comment, perhaps something like, "The target tool should only have been cleared if one of the target groups is in its path and if none of the target group values is empty."  Ditto for remotes and tools.